bigtable/client/row_range.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BIGTABLE_CLIENT_ROW_RANGE_H_
+#define BIGTABLE_CLIENT_ROW_RANGE_H_
+
+#include <iostream>
+#include <string>
+
+#include "bigtable/client/version.h"
+
+#include <google/bigtable/v2/data.pb.h>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+class RowRange {
+ public:
+  RowRange(RowRange&& rhs) noexcept = default;
+  RowRange& operator=(RowRange&& rhs) noexcept = default;
+  RowRange(RowRange const& rhs) = default;
+  RowRange& operator=(RowRange const& rhs) = default;
+  RowRange() = default;
+  RowRange(std::string begin, std::string end);
+
+  // Create different types of ranges.
+  // A range with 'end' not set; 'start' might be either set or not set.
```

Comment:
Please use Doxygen style comments (`/** ... */`) in the header, this is where the automatically generated documentation is extracted from.  See [coryan.github.io/google-cloud-cpp](https://coryan.github.io/google-cloud-cpp) for the resulting output.